pmd-ant/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -37,7 +35,6 @@
 import net.sourceforge.pmd.reporting.GlobalAnalysisListener;
 import net.sourceforge.pmd.reporting.ListenerInitializer;
 
-@InternalApi
 public class Formatter {
```

Comment:
Can we move that to the internal package? That way we wouldn't need the internal API bridge class

Reply:
I consider this class as public API, as this is used to configure the PMD Task:
e.g. https://docs.pmd-code.org/apidocs/pmd-ant/7.0.0-SNAPSHOT/net/sourceforge/pmd/ant/PMDTask.html#addFormatter(net.sourceforge.pmd.ant.Formatter)

This is basically, what ant calls, when it sees this in xml:

```xml
<pmd ...>
    <formatter type="html" toFile="${build}/pmd_report.html"/>
</pmd>
```

Well, it creates a new instance of Formatter, calls [setType](https://docs.pmd-code.org/apidocs/pmd-ant/7.0.0-SNAPSHOT/net/sourceforge/pmd/ant/Formatter.html#setType(java.lang.String)) on it and [setToFile](https://docs.pmd-code.org/apidocs/pmd-ant/7.0.0-SNAPSHOT/net/sourceforge/pmd/ant/Formatter.html#setToFile(java.io.File)), and then calls `addFormatter` on the PMDTask instance.

I think, the Formatter should really only be a data class without any logic (it's part of the configuration of the ant task). Right now, it seems to do a bit more (like creating a renderer by itself - which should really be part of the implementation PMDTaskImpl...). But that's something to refactor later.

Having these InternalApiBridge classes is kind of a compromise I guess. Not nice, but does the job of hiding methods in public classes.

Reply:
Ok thanks for explaining, that makes sense. Probably this should be mentioned in a comment in the code